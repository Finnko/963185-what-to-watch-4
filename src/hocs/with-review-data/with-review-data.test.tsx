import * as React from 'react';
import * as renderer from 'react-test-renderer';
import RadioButton from '../../components/radio-button/radio-button';
import withReviewData from './with-review-data';

type MockComponentProps = {
  review: string;
  rating: string;
  isFormValid: boolean;
  onInputChange: () => void;
}

const MockComponent = (props: MockComponentProps) => {
  const {
    review,
    rating,
    isFormValid,
    onInputChange,
  } = props;

  return (
    <div>
      <div className="rating__stars">
        <RadioButton
          key={1}
          id={`star-1`}
          groupName="rating"
          checked={rating === `1`}
          value="1"
          label={`Rating 1`}
          disabled={false}
          onRadioChange={onInputChange}
        />
      </div>
      <textarea
        className="add-review__textarea"
        name="review"
        id="review"
        placeholder="Review text"
        value={review}
        onChange={onInputChange}
      />
      <button
        className="add-review__btn"
        type="submit"
        disabled={!isFormValid}
      >
        Post
      </button>
    </div>
  );
};

const MockComponentWrapped = withReviewData(MockComponent);

describe(`Should hoc withReviewData render correctly`, () => {
  it(`Snapshot render withReviewData`, () => {
    const tree = renderer.create(<MockComponentWrapped />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
