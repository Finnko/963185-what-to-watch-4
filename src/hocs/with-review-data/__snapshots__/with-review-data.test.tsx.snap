// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should hoc withReviewData render correctly Snapshot render withReviewData 1`] = `
<div>
  <div
    className="rating__stars"
  >
    <input
      checked={false}
      className="rating__input"
      disabled={false}
      id="star-1"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <label
      className="rating__label"
      htmlFor="star-1"
    >
      Rating 1
    </label>
  </div>
  <textarea
    className="add-review__textarea"
    id="review"
    name="review"
    onChange={[Function]}
    placeholder="Review text"
    value=""
  />
  <button
    className="add-review__btn"
    disabled={true}
    type="submit"
  >
    Post
  </button>
</div>
`;
